src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -701,22 +564,22 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     TerminalConnection::ConnectionState TermControl::ConnectionState() const
     {
-        return _connection.State();
+        return _core->ConnectionState();
     }
 
-    winrt::fire_and_forget TermControl::RenderEngineSwapChainChanged()
+    winrt::fire_and_forget TermControl::RenderEngineSwapChainChanged(const IInspectable& /*sender*/, const IInspectable& /*args*/)
     {
         // This event is only registered during terminal initialization,
         // so we don't need to check _initializedTerminal.
         // We also don't lock for things that come back from the renderer.
-        auto chain = _renderEngine->GetSwapChain();
+        auto chain = _core->GetSwapChain();

```

Comment:
hmm. did the type of `chain` change here? why do we not need to `Get`?

If we were using a smart pointer type before, it was guaranteeing that there was a +1 reference count on the chain. Now if we're not, the object could die between the co_await and the resume.

Reply:
Yea I switched to the raw ptr, because I didn't want to futz with the WRL headers. Moving inside the coroutine though is trivial